src/components/ConnectionSidePanel.jsx

```diff
@@ -1,36 +1,36 @@
-import React, {useContext} from 'react';
+import {useContext} from 'react';
 import {useTranslation} from 'react-i18next';
 import {useDispatch} from 'react-redux';
 
 import api from '../api/apiInstance';
+import {webError} from './../utilities/messagesHelper';
 import {basicHeaders} from '../config/apiConfig';
 import {generateConnectionData, generateStopName} from '../utilities/mapUtilities';
-import {unsafeApiError} from '../utilities/utilities';
 import {MapContext} from './contexts/MapContextProvider';
 import {NotificationActions} from '../redux/actions/notificationActions';
 
 import '../stylesheets/connectionPrompt.scss';
 
 const ConnectionSidePanel = () => {
-  const {connectionData, reset, shouldRenderConnections} = useContext(MapContext);
+  const {connectionData, reset, shouldRenderConnections, activeTile} = useContext(MapContext);
   const {t} = useTranslation();
   const dispatch = useDispatch();
 
   const sendConnection = async () => {
-    if (connectionData.length !== 2) {
-      dispatch(NotificationActions.error(t('connection.mark2Stops')));
-      return;
-    }
     try {
-      await api.connectionsUpdate(generateConnectionData(connectionData), {
+      if (connectionData.length !== 2) {
+        dispatch(NotificationActions.error(t('connection.mark2Stops')));
+        return;
+      }
+
+      await api.connectionsUpdate(generateConnectionData(connectionData, activeTile.id), {
         headers: basicHeaders(),
       });
       shouldRenderConnections(true);
       reset();
       dispatch(NotificationActions.success(t('connection.createSuccessMessage')));
     } catch (error) {
-      unsafeApiError(error);
-      dispatch(NotificationActions.error(error.errors.message & error.errors.message[0] ? error.errors.message[0] : t('unrecognizedProblem')));
+      error instanceof Response ? webError(error) : dispatch(NotificationActions.error(t('error.exception')));
```

Comment:
when error won't be instance of Response object?

Reply:
User will be informed about it with generic message: `dispatch(NotificationActions.error(t('error.exception'))`.